src/main/java/com/fasterxml/jackson/databind/deser/SettableAnyProperty.java

```diff
@@ -141,8 +145,31 @@ public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
     public void set(Object instance, String propName, Object value) throws IOException
     {
         try {
-            // note: can not use 'setValue()' due to taking 2 args
-            _setter.getAnnotated().invoke(instance, propName, value);
+			// note: can not use 'setValue()' due to taking 2 args
+			if (_setter instanceof AnnotatedMethod) {
+				((AnnotatedMethod) _setter).getAnnotated().invoke(instance, propName, value);
+			}
+			// if annotation in the field (only map is supported now)
+			else if (_setter instanceof AnnotatedField) {
+				AnnotatedField field = ((AnnotatedField) _setter);
+				// check whether the annotated field is an instance of Map, if
+				// not throw Error
+				if (!Map.class.isAssignableFrom(field.getRawType())) {
+					throw new IllegalArgumentException(
+					        "Expecting a java.util.Map type in JsonAnySetter field, but got " + field.getRawType());
+				}
+				// get the field value, if null instantiate it..
+				Object val = field.getValue(instance);
+				if (val == null) {
+					val = new HashMap();
```

Comment:
This is something that I am not 100% sure about. I can see the benefit (and need), to some degree, but it is also possible that user specifies different nominal subtype (like `TreeMap` or `ConcurrentHashMap`), in which case assignment will fail. 

So one possibility would be to simply skip assignment if field has `null`; otherwise I am not 100% how to avoid problem cases.